docs/framework-overview.md

```diff
@@ -0,0 +1,398 @@
+bigger toolchains (*e.g.* the FOSS toolchain). Therefore, the resulting options
+of the toolchain, its attributes and methods will be the combined result of all
+inherited class components.
+
+```mermaid
+flowchart
+    toolchain --> compiler & mpi & linalg & fft
+    subgraph "Compiler"
+        compiler --> gcc --> gcccore --> gcctoolchain 
+    end
+    subgraph "MPI Implementation"
+        mpi --> ompi 
+    end
+    gcctoolchain & ompi --> gompi --> foss
+    subgraph "Fast-Fourier Libraries"
+        fft --> fftw 
+    end
+    fftw --> foss
+    subgraph "Linear Algebra Libraries"
+        linalg --> oblas & fblas
+        linalg --> blacs --> scalapack
+    end
+    oblas & fblas & scalapack --> foss
+    toolchain("class Toolchain
+    eb.tools.toolchain.toolchain")
+    compiler("class Compiler
+    eb.tools.toolchain.compiler")
+    gcc("class Gcc
+    eb.toolchains.compiler.gcc")
+    gcccore("class GccCore
+    eb.toolchains.gcccore")
+    gcctoolchain("class GccToolchain
+    eb.toolchains.gcc")
+    mpi("class Mpi
+    eb.tools.toolchain.mpi")
+    ompi("class OpenMPI
+    eb.toolchains.mpi.openmpi")
+    gompi("class Gompi
+    eb.toolchains.gompi")
+    fft("class Fft
+    eb.tools.toolchain.fft")
+    fftw("class Fftw
+    eb.toolchains.fft.fftw")
+    linalg("class LinAlg
+    eb.tools.toolchain.linalg")
+    oblas("class OpenBLAS
+    eb.toolchains.linalg.openblas")
+    fblas("class FlexiBLAS
+    eb.toolchains.linalg.flexiblas")
+    blacs("class Blacs
+    eb.toolchains.linalg.blacs")
+    scalapack("class ScaLAPACK
+    eb.toolchains.linalg.scalapack")
+    foss("class Foss
+    eb.toolchains.foss")
+```
+
+## Features in tools
+
+### EasyBuild Configuration
+
+Configuration options in EasyBuild can come from multiple sources:
+
+* command line arguments to the `eb` command
+* environment variables
+* settings in configuration files
+* settings in easyconfig files
+
+!!! tip
+    Check the [Configuring EasyBuild][configuring_easybuild] for details on how
+    to use the different configuration mechanisms.
+
+All configuration options in EasyBuild are defined in `easybuild.tools.config`.
+This includes options available through the `eb` command and those that are
+not. Options that also have a companion command line argument must define the
+name of the argument and its help string in the `EasyBuildOptions` class of
+`easybuild.tools.options`. Therefore, adding a new option to `eb` is a two step
+process affecting the `config` and `options` modules.
+
+The `easybuild.tools.options` module also handles the parsing of options through its `set_up_configuration()` method, which is called in the [very early stages][workflow_overview] of the `eb` execution. The main steps are `parse_options()` which parses the command line itself, `process_software_build_specs()` which processes the build options found in the easyconfig file and `parse_external_modules_metadata()` which parses external configuration files.
```

Comment:
I'm not sure we should even mention `process_software_build_specs` and `parse_external_modules_metadata`, because those are only relevant for fairly obscure features.

The latter is for https://docs.easybuild.io/using-external-modules/#using_external_modules_metadata

The former is actually for a rather weird way to specify specific easyconfig parameters in the `eb` command, without specifying an easyconfig file at all (EasyBuild will try to find an easyconfig file based on what's specified).
That's for options like `--software-name`, `--software-version`, which nobody ever uses afaik...

Reply:
I misunderstood the role of this functions. Fixed.